### ISSUE TITLE
Error when using environmental RasterLayer and not RasterStack

### ISSUE THREAD

**Issue by Rekyt** [sylvainschmitt/SSDM#20]
Hi @sylvainschmitt ,
Thank you for building SSDM!
Because I'm working on simple simulations with a single variable I discovered a simple bug: when using a `RasterLayer` as the `Env` variable in the function `modelling()` I got an error.

```r
Error in modelling("GLM", occ, env_raster, Xcol = "x", Ycol = "y", Pcol = "presence") : 
  Environment data set is not a raster or a raster stack
```

Here is a minimal reprex:
<details><summary>Minimal Reproductible Example</summary>

```r

library("SSDM")
#> Welcome to the SSDM package, you can launch the graphical user interface by typing gui() in the console.
library("raster")
#> Le chargement a nÃ©cessitÃ© le package : sp

env_raster = raster(matrix(1:10))  # Create simple environmental gradient

occ = data.frame(presence = c(rep(1, 9), 0), cell = 1:10)  # Dummy occurences

# Extract cell coordinates
cell_xy = xyFromCell(env_raster, occ$cell)
occ$x = cell_xy[, "x"]
occ$y = cell_xy[, "y"]

# Not working model ###
raster_layer = modelling("GLM", occ, env_raster, Xcol = "x", Ycol = "y",
                         Pcol = "presence")
#> Data check ...
#> Error in modelling("GLM", occ, env_raster, Xcol = "x", Ycol = "y", Pcol = "presence"): Environment data set is not a raster or a raster stack

# Working model ####
raster_stack = modelling("GLM", occ, stack(env_raster), Xcol = "x", Ycol = "y",
                         Pcol = "presence")
#> Data check ... 
#>    done. 
#> 
#> Model evaluation...
#>    done. 
#> 
#> Model projection...
#>    done. 
#> 
#> Model axes contribution evaluation...
#>    done.

```

</details>

**My workaround** is currently to wrap my `RasterLayer` into a call of `stack()`.
But it would great if it wouldn't be necessary when you only need a single environmental variable.

**Possible bug origin**: I think the problem comes from https://github.com/sylvainschmitt/SSDM/blob/971795bbc8f6694c736a5003557d0c775007b840/R/modelling.R#L298

It could be replaced by `if (!inherits(Env, "Raster"))` I'm going to check if it doesn't break anything before making a PR ;)

**Comment by Rekyt** [comment]
If you want to have a look, I did something at https://github.com/Rekyt/SSDM/tree/fix_raster_layer

Sidenote: I added `context()` to all tests to fix r-lib/testthat#700

I'm having issues currently with https://github.com/sylvainschmitt/SSDM/blob/971795bbc8f6694c736a5003557d0c775007b840/R/data.values.R#L10
Because it creates weird variable names for `RasterLayer` such as `extract.Env..cbind.obj.data.X..obj.data.Y..` and seems to make `modelling()` fail.

**Comment by sylvainschmitt** [comment]
There was supposed to be a security for this case that does not seems to work:
https://github.com/sylvainschmitt/SSDM/blob/939b53e6fea3d187c1c558f97e8784627526548b/R/modelling.R#L295-L297
I will fix it but anyway I would not much use an SDM with only one predictor.
